java/vector/src/main/java/org/apache/arrow/vector/GenerateSampleData.java

```diff
@@ -267,6 +275,58 @@ private static void writeTinyIntData(TinyIntVector vector, int valueCount) {
     vector.setValueCount(valueCount);
   }
 
+  private static void writeUInt1Data(UInt1Vector vector, int valueCount) {
+    final byte even = 1;
+    final byte odd = 2;
+    for (int i = 0; i < valueCount; i++) {
+      if (i % 2 == 0) {
+        vector.setSafe(i, even);
+      } else {
+        vector.setSafe(i, odd);
+      }
+    }
+    vector.setValueCount(valueCount);
+  }
+
+  private static void writeUInt2Data(UInt2Vector vector, int valueCount) {
+    final short even = 10;
+    final short odd = 20;
+    for (int i = 0; i < valueCount; i++) {
+      if (i % 2 == 0) {
+        vector.setSafe(i, even);
+      } else {
+        vector.setSafe(i, odd);
+      }
+    }
+    vector.setValueCount(valueCount);
+  }
+
+  private static void writeUInt4Data(UInt4Vector vector, int valueCount) {
+    final int even = 1000;
+    final int odd = 2000;
+    for (int i = 0; i < valueCount; i++) {
+      if (i % 2 == 0) {
+        vector.setSafe(i, even);
+      } else {
+        vector.setSafe(i, odd);
+      }
+    }
+    vector.setValueCount(valueCount);
+  }
+
+  private static void writeUInt8Data(UInt8Vector vector, int valueCount) {
+    final long even = 1000000000;
+    final long odd = 2000000000;
```

Comment:
Is it possible to use a value that wouldn't be representable in the signed equivalent of the vectors? (Not sure how you write such a value though)

Reply:
The UInts are all represented as ints under the covers in java (except UInt2 may be a char). I don't know of any way to to create a value like you're suggesting, and I think it would error if I did so it couldn't be used for test data.

Reply:
Well, the value has to be representable somehow (else how would IPC work? I'm guessing you'd just take a negative int and it would reinterpret it as an unsigned integer) and I'd want us to test a range of values that may occur (the C++ unit tests for kernels, for instance, try to test the extremes, 0, negatives, etc. to catch corner cases, and frameworks like Hypothesis generalize this)